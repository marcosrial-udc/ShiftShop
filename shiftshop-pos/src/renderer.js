import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {Provider} from 'react-redux';
import {MuiThemeProvider} from '@material-ui/core';
import {SnackbarProvider} from 'notistack';
import {IntlProvider} from 'react-intl';

/* Async/await regenerator necesaries */
import 'core-js/stable';
import 'regenerator-runtime/runtime';

import './assets/scss/index.scss';
import theme from './theme'

import {App} from './modules/app';

import configureStore from './store';
import {initReactIntl} from "./i18n";
import './polyfills';

/* Configure store. */
const store = configureStore();

/* Configure i18n. */
const {locale, messages} = initReactIntl();

ReactDOM.render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <IntlProvider locale={locale} messages={messages}>
          <SnackbarProvider anchorOrigin={{vertical: 'top', horizontal: 'center'}}
                            autoHideDuration={2500}>
            <App/>
          </SnackbarProvider>
        </IntlProvider>
      </Provider>
    </MuiThemeProvider>,
    document.getElementById("root"));
