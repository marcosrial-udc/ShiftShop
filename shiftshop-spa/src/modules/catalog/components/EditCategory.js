import React, {useEffect, useRef, useState} from 'react';
import {useDispatch} from 'react-redux';
import {FormattedMessage} from 'react-intl';
import {useSnackbar} from 'notistack';
import PropTypes from 'prop-types';
import {
    Button,
    CircularProgress,
    Dialog,
    DialogActions,
    DialogContent,
    DialogTitle,
    TextField, Typography
} from '@material-ui/core';

import useStyles from '../styles/EditCategory';

import * as actions from '../actions';
import {Alert, ErrorContent} from '../../common';

const EditCategory = ({category, onClose}) => {
    const classes = useStyles();

    const dispatch = useDispatch();
    const {enqueueSnackbar, closeSnackbar} = useSnackbar();

    const _isMounted = useRef(true);
    useEffect(() => {
        return () => _isMounted.current = false
    }, []);

    const [name, setName] = useState('');
    const [updating, setUpdating] = useState(false);
    const [errors, setErrors] = useState(null);

    const closeErrors = () => setErrors(null);

    const checkValid = name.trim() !== '';

    const handleSubmit = e => {

        e.preventDefault();
        closeErrors();

        if (checkValid && !updating) {
            editCategory();
        }
    };

    const editCategory = () => {

        if (updating) return;

        setUpdating(true);

        dispatch(
            actions.updateCategory(
                category.id,
                {name: name.trim()},
                _ => {
                    enqueueSnackbar(<FormattedMessage id="project.catalog.EditCategory.success"/>,
                        {
                            variant: 'success',
                            autoHideDuration: 1500,
                        });
                    onClose();
                },
                errors => {
                    if (_isMounted.current) {
                        setErrors(errors)
                    } else {
                        enqueueSnackbar(<ErrorContent errors={errors}/>,
                            {
                                variant: 'error',
                                persist: 'true',
                                action: key => (<Button color="inherit" variant="outlined" onClick={() => closeSnackbar(key)}>
                                    <FormattedMessage id="project.global.button.close"/>
                                </Button>)
                            });
                    }
                },
                () => _isMounted.current && setUpdating(false)));

    };

    if (!category || !onClose) {
        return null;
    }

    return (
        <Dialog open maxWidth="xs" fullWidth onClose={onClose}>
            <DialogTitle>
                <Typography component="span" variant="h5">
                    <FormattedMessage id="project.catalog.EditCategory.title"/>: <i>{category.name}</i>
                </Typography>
            </DialogTitle>
            <form onSubmit={e => handleSubmit(e)} noValidate>
                <DialogContent dividers>
                    {errors && <Alert className={classes.alert} variant="error" message={errors} onClose={closeErrors} backendError/>}
                    <TextField autoFocus margin="dense" type="text" variant="outlined" fullWidth
                               label={<FormattedMessage id="project.global.field.name"/>}
                               value={name} onChange={e => setName(e.target.value)}/>
                </DialogContent>
                <DialogActions>
                    <Button variant="contained" size="small" color="default" onClick={onClose} disableElevation>
                        <FormattedMessage id="project.global.button.cancel"/>
                    </Button>
                    <Button variant="contained" size="small" color="primary" type="submit" disableElevation
                            disabled={!checkValid || updating}>
                        <FormattedMessage id="project.global.button.accept"/>
                        {updating && <CircularProgress className={classes.buttonProgress} size={24}/>}
                    </Button>
                </DialogActions>
            </form>
        </Dialog>
    );
};

EditCategory.propTypes = {
    category: PropTypes.object.isRequired,
    onClose: PropTypes.func.isRequired
};

export default EditCategory;
