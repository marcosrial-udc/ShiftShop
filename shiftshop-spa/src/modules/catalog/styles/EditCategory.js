import {makeStyles} from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    root: {},
    alert: {marginBottom: theme.spacing(2)},
    buttonProgress: {position: 'absolute'}
}));

export default useStyles;
