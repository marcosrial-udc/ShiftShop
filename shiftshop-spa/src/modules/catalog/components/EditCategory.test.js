import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {SnackbarProvider} from 'notistack';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl';

import {initReactIntl} from "../../../i18n";

import EditCategory from './EditCategory';

const {messages} = initReactIntl();

describe('EditCategory snapshot', () => {

    const middlewares = [];
    const mockStore = configureStore(middlewares);

    const createEditCateogry = props => {

        const store = mockStore();

        return renderer.create(
            <Provider store={store}>
                <IntlProvider locale='en' messages={messages}>
                    <SnackbarProvider>
                        <EditCategory {...props}/>
                    </SnackbarProvider>
                </IntlProvider>
            </Provider>
        );

    };

    const categoryExample = {id: 1, name: 'Category'};

    test('EditCategory with null category', () => {

        const editCategory = createEditCateogry({category: null, onClose: jest.fn()});

        expect(editCategory.toJSON()).toMatchSnapshot();

    });

    test('EditCategory with null onClose', () => {

        const editCategory = createEditCateogry({category: categoryExample, onClose: null});

        expect(editCategory.toJSON()).toMatchSnapshot();

    });

    /* Error trying to run with Dialog components -> material-ui Fade component test issue

    test('EditCategory with correct props', () => {

        const editCategory = createEditCateogry({category: categoryExample, onClose: jest.fn()});

        expect(editCategory.toJSON()).toMatchSnapshot();

    });

    */

});
