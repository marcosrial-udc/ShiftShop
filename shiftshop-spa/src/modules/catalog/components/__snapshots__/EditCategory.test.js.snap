// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCategory snapshot EditCategory with null category 1`] = `null`;

exports[`EditCategory snapshot EditCategory with null onClose 1`] = `null`;
